pkg/kapp/preflight/registry_test.go

```diff
@@ -102,7 +119,111 @@ func TestRegistryRun(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			err := tc.registry.Run(nil, nil)
-			require.Equal(t, tc.shouldErr, err != nil)
+			require.Equalf(t, tc.shouldErr, err != nil, "Unexpected error: %v", err)
+		})
+	}
+}
+
+func TestRegistryConfig(t *testing.T) {
+	testCases := []struct {
+		name       string
+		registry   *Registry
+		configYaml string
+		shouldErr  bool
+	}{
+		{
+			name: "preflight checks registered, config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(
+						nil,
+						func(cfg CheckConfig) error {
+							if cfg == nil {
+								return errors.New("config should be present")
+							}
+							v, ok := cfg["foo"]
+							if !ok {
+								return errors.New("foo config not present")
+							}
+							if v != "bar" {
+								return errors.New("foo should equal 'bar'")
+							}
+							_, ok = cfg["foobar"]
+							if ok {
+								return errors.New("foobar config should not present")
+							}
+							return nil
+						},
+						true,
+					),
+				},
+			},
+			configYaml: `---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+preflightRules:
+- name: someCheck
+  config:
+    foo: bar
+`,
+		},
+		{
+			name: "preflight checks registered, unexpected preflight check set, error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(
+						nil,
+						func(cfg CheckConfig) error {
+							if cfg != nil {
+								return errors.New("config should not be present")
+							}
+							return nil
+						},
+						true,
+					),
+				},
+			},
+			configYaml: `---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+preflightRules:
+- name: otherCheck
+  config:
+    foo: bar
+`,
+			shouldErr: true,
+		},
+		{
+			name: "preflight checks registered, duplicate config entry, error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(nil, nil, true),
+				},
+			},
+			configYaml: `---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+preflightRules:
+- name: someCheck
+  config:
+    foo: bar
+- name: someCheck
+  config:
+    bar: foo
+`,
+			shouldErr: true,
+		},
+	}
```

Comment:
Correct me if I am wrong, but it looks like we have logic for the following scenarios:
- Preflight checks are registered, unknown preflight check is in the config, error
- Preflight checks are registered, duplicate preflight check is in the config, error (has test)
- Preflight checks are registered, checks enabled via flag, checks are enabled/disabled according to the flag
- Preflight checks are registered, checks not enabled via flag, checks are enabled via the configuration file
- Preflight checks are registered, setting configuration for a check fails, error (has test)
- Preflight checks are registered, setting configuration succeeds for all checks, no error (has test)

Could we add some additional tests to cover the scenarios that are not already covered above?